crates/libcontainer/src/hooks.rs

```diff
@@ -27,16 +28,35 @@ pub enum HookError {
     MissingContainerState,
     #[error("failed to write container state to stdin")]
     WriteContainerState(#[source] std::io::Error),
+    #[error("io error")]
+    OtherIO(#[source] std::io::Error),
 }
 
 type Result<T> = std::result::Result<T, HookError>;
 
-pub fn run_hooks(hooks: Option<&Vec<Hook>>, container: Option<&Container>) -> Result<()> {
+pub fn run_hooks(
+    hooks: Option<&Vec<Hook>>,
+    container: Option<&Container>,
+    cwd: Option<&Path>,
+) -> Result<()> {
     let state = &(container.ok_or(HookError::MissingContainerState)?.state);
 
     if let Some(hooks) = hooks {
         for hook in hooks {
-            let mut hook_command = process::Command::new(hook.path());
+            let mut cmd = process::Command::new(hook.path());
+
+            let hook_command = if let Some(cwd) = cwd {
+                cmd.current_dir(cwd)
+            } else {
+                cmd.current_dir(
+                    env::current_dir()
+                        .map_err(|err| {
+                            tracing::error!("failed to get current directory: {}", err);
+                            HookError::OtherIO(err)
+                        })?
+                        .as_path(),
+                )
+            };
```

Comment:
I don't think the else section is needed, as by default cmd is run in the calling process's directory: ref - third example in the docs [here](https://doc.rust-lang.org/std/process/struct.Command.html)

Reply:
Initially, I want to ensure cmd works fine even if the execution of `chdir` after this line.  But I thought about it again, we can notice the cover in the unite test, so I'll turn it off.

Reply:
73cffe381028c034686d042a6e11de9649a59a3b